third_party/2and3/yaml/__init__.pyi

```diff
@@ -20,12 +20,49 @@ def load_all(stream: Union[str, IO[str]], Loader=...) -> Iterator[Any]: ...
 def safe_load(stream: Union[str, IO[str]]) -> Any: ...
 def safe_load_all(stream: Union[str, IO[str]]) -> Iterator[Any]: ...
 def emit(events, stream=..., Dumper=..., canonical=..., indent=..., width=..., allow_unicode=..., line_break=...): ...
-def serialize_all(nodes, stream=..., Dumper=..., canonical=..., indent=..., width=..., allow_unicode=..., line_break=..., encoding=..., explicit_start=..., explicit_end=..., version=..., tags=...): ...
-def serialize(node, stream=..., Dumper=..., **kwds): ...
-def dump_all(documents: Sequence[Any], stream: IO[str] = ..., Dumper=..., default_style=..., default_flow_style=..., canonical=..., indent=..., width=..., allow_unicode=..., line_break=..., encoding=..., explicit_start=..., explicit_end=..., version=..., tags=...) -> Any: ...
-def dump(data: Any, stream: IO[str] = ..., Dumper=..., **kwds) -> Any: ...
-def safe_dump_all(documents: Sequence[Any], stream: IO[str] = ..., **kwds) -> Any: ...
-def safe_dump(data: Any, stream: IO[str] = ..., **kwds) -> Any: ...
+
+@overload
+def serialize_all(nodes, stream: None=..., Dumper=..., canonical=..., indent=..., width=..., allow_unicode=..., line_break=..., encoding: None=..., explicit_start=..., explicit_end=..., version=..., tags=...) -> str: ...
+@overload
+def serialize_all(nodes, stream: None=..., Dumper=..., canonical=..., indent=..., width=..., allow_unicode=..., line_break=..., encoding: str=..., explicit_start=..., explicit_end=..., version=..., tags=...) -> bytes: ...
```

Comment:
Mypy complains about this because both overloads have `= ...` for the `encoding` argument. One of the two overloads should have no default for `encoding`.

Reply:
Until #5621 is fixed I see no practical way to resolve this: `encoding` is in the middle of 11 optional parameters and using that workaround would require 2^6 overload variants of those 6 functions.
I'm out of ideas for now.